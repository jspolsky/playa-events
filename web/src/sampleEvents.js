export const eventsAnahasanaVillage = [
  {
    start: new Date(2021, 7, 29, 18, 0, 0, 0),
    end: new Date(2021, 7, 29, 19, 30, 0, 0),
    title: "Evening Ecstatic Dance",
  },
  {
    start: new Date(2021, 7, 30, 18, 0, 0, 0),
    end: new Date(2021, 7, 30, 19, 30, 0, 0),
    title: "Evening Ecstatic Dance",
  },
  {
    start: new Date(2021, 7, 31, 18, 0, 0, 0),
    end: new Date(2021, 7, 31, 19, 30, 0, 0),
    title: "Evening Ecstatic Dance",
  },
  {
    start: new Date(2021, 8, 1, 18, 0, 0, 0),
    end: new Date(2021, 8, 1, 19, 30, 0, 0),
    title: "Evening Ecstatic Dance",
  },
  {
    start: new Date(2021, 8, 2, 18, 0, 0, 0),
    end: new Date(2021, 8, 2, 19, 30, 0, 0),
    title: "Evening Ecstatic Dance",
  },
  {
    start: new Date(2021, 8, 3, 18, 0, 0, 0),
    end: new Date(2021, 8, 3, 19, 30, 0, 0),
    title: "Evening Ecstatic Dance",
  },
  {
    start: new Date(2021, 7, 30, 7, 30, 0),
    end: new Date(2021, 7, 30, 8, 30, 0),
    title: "Morning Meditation",
  },
  {
    start: new Date(2021, 7, 31, 7, 30, 0),
    end: new Date(2021, 7, 31, 8, 30, 0),
    title: "Morning Meditation",
  },
  {
    start: new Date(2021, 8, 1, 7, 30, 0),
    end: new Date(2021, 8, 1, 8, 30, 0),
    title: "Morning Meditation",
  },
  {
    start: new Date(2021, 8, 2, 7, 30, 0),
    end: new Date(2021, 8, 2, 8, 30, 0),
    title: "Morning Meditation",
  },
  {
    start: new Date(2021, 8, 3, 7, 30, 0),
    end: new Date(2021, 8, 3, 8, 30, 0),
    title: "Morning Meditation",
  },
  {
    start: new Date(2021, 8, 4, 7, 30, 0),
    end: new Date(2021, 8, 4, 8, 30, 0),
    title: "Morning Meditation",
  },

  ///

  {
    start: new Date(2021, 7, 30, 18, 0, 0),
    end: new Date(2021, 7, 30, 19, 0, 0),
    title: "Basic Income In Depth",
  },
  {
    start: new Date(2021, 7, 31, 11, 0, 0),
    end: new Date(2021, 7, 31, 12, 0, 0),
    title: "Decentralized Permaculture Networks",
  },
  {
    start: new Date(2021, 8, 1, 13, 0, 0),
    end: new Date(2021, 8, 1, 14, 0, 0),
    title: "We Can Do Better as a Species",
  },
  {
    start: new Date(2021, 8, 2, 15, 0, 0),
    end: new Date(2021, 8, 2, 16, 0, 0),
    title: "Decentralize Your Healthcare!",
  },

  {
    start: new Date(2021, 8, 3, 11, 0, 0),
    end: new Date(2021, 8, 3, 12, 0, 0),
    title: "Modern Cybernetics",
  },
  {
    start: new Date(2021, 8, 3, 12, 0, 0),
    end: new Date(2021, 8, 3, 13, 0, 0),
    title: "Blockchain4Good! Saving the world!",
  },
  {
    start: new Date(2021, 7, 30, 12, 0, 0),
    end: new Date(2021, 7, 30, 13, 0, 0),
    title: "Intro to Blockchain Discussion",
  },
  {
    start: new Date(2021, 7, 31, 12, 0, 0),
    end: new Date(2021, 7, 31, 13, 0, 0),
    title: "Intro to Blockchain Discussion",
  },
  {
    start: new Date(2021, 8, 1, 12, 0, 0),
    end: new Date(2021, 8, 1, 13, 0, 0),
    title: "Intro to Blockchain Discussion",
  },
  {
    start: new Date(2021, 8, 2, 12, 0, 0),
    end: new Date(2021, 8, 2, 13, 0, 0),
    title: "Intro to Blockchain Discussion",
  },

  {
    start: new Date(2021, 7, 30, 15, 30, 0),
    end: new Date(2021, 7, 30, 17, 0, 0),
    title: "Intro to Blockchain Talk with Q&A",
  },
  {
    start: new Date(2021, 7, 31, 15, 30, 0),
    end: new Date(2021, 7, 31, 17, 0, 0),
    title: "Intro to Blockchain Talk with Q&A",
  },
  {
    start: new Date(2021, 8, 1, 15, 30, 0),
    end: new Date(2021, 8, 1, 17, 0, 0),
    title: "Intro to Blockchain Talk with Q&A",
  },
  {
    start: new Date(2021, 8, 2, 15, 30, 0),
    end: new Date(2021, 8, 2, 17, 0, 0),
    title: "Intro to Blockchain Talk with Q&A",
  },
  {
    start: new Date(2021, 8, 3, 15, 30, 0),
    end: new Date(2021, 8, 3, 17, 0, 0),
    title: "Intro to Blockchain Talk with Q&A",
  },
  {
    start: new Date(2021, 7, 31, 18, 0, 0),
    end: new Date(2021, 7, 31, 19, 0, 0),
    title: "Rising Coffee Consciousness & Ceremony",
  },
  {
    start: new Date(2021, 8, 3, 18, 0, 0),
    end: new Date(2021, 8, 3, 19, 0, 0),
    title: "Rising Coffee Consciousness & Ceremony",
  },
  {
    start: new Date(2021, 7, 30, 8, 30, 0),
    end: new Date(2021, 7, 30, 10, 0, 0),
    title: "Morning Yoga",
  },
  {
    start: new Date(2021, 7, 31, 8, 30, 0),
    end: new Date(2021, 7, 31, 10, 0, 0),
    title: "Morning Yoga",
  },
  {
    start: new Date(2021, 8, 1, 8, 30, 0),
    end: new Date(2021, 8, 1, 10, 0, 0),
    title: "Morning Yoga",
  },
  {
    start: new Date(2021, 8, 2, 8, 30, 0),
    end: new Date(2021, 8, 2, 10, 0, 0),
    title: "Morning Yoga",
  },
  {
    start: new Date(2021, 8, 3, 8, 30, 0),
    end: new Date(2021, 8, 3, 10, 0, 0),
    title: "Morning Yoga",
  },
  {
    start: new Date(2021, 8, 4, 8, 30, 0),
    end: new Date(2021, 8, 4, 10, 0, 0),
    title: "Morning Yoga",
  },
  {
    start: new Date(2021, 7, 30, 10, 30, 0),
    end: new Date(2021, 7, 30, 12, 0, 0),
    title: "Morning Ecstatic Dance",
  },
  {
    start: new Date(2021, 7, 31, 10, 30, 0),
    end: new Date(2021, 7, 31, 12, 0, 0),
    title: "Morning Ecstatic Dance",
  },
  {
    start: new Date(2021, 8, 1, 10, 30, 0),
    end: new Date(2021, 8, 1, 12, 0, 0),
    title: "Morning Ecstatic Dance",
  },
  {
    start: new Date(2021, 8, 2, 10, 30, 0),
    end: new Date(2021, 8, 2, 12, 0, 0),
    title: "Morning Ecstatic Dance",
  },
  {
    start: new Date(2021, 8, 3, 10, 30, 0),
    end: new Date(2021, 8, 3, 12, 0, 0),
    title: "Morning Ecstatic Dance",
  },
  {
    start: new Date(2021, 8, 4, 10, 30, 0),
    end: new Date(2021, 8, 4, 12, 0, 0),
    title: "Morning Ecstatic Dance",
  },
];

export const events = [
  {
    start: new Date(2021, 7, 30, 20, 0, 0),
    end: new Date(2021, 7, 31, 2, 0, 0),
    title: "Emergence: 3907's Birth Day Party",
    description:
      "BRC Summit PARTY, come elevate your senses 3907 feet above sea level.",
  },
  {
    start: new Date(2021, 7, 31, 14, 0, 0),
    end: new Date(2021, 7, 31, 16, 0, 0),
    title: "T Shirt printing",
    description: "Come stamp different BRC patterns onto fabric, byof",
  },
  {
    start: new Date(2021, 7, 31, 18, 0, 0),
    end: new Date(2021, 7, 31, 19, 0, 0),
    title: "Elevate the Latino within",
    description: "Caipirinhas and Brazilian Music",
  },
  {
    start: new Date(2021, 8, 1, 14, 0, 0),
    end: new Date(2021, 8, 1, 15, 0, 0),
    title: "Tune in to Touch",
    description:
      "Learn deep and relaxing massage techniques. Hands on with a partner.",
  },
  {
    start: new Date(2021, 8, 1, 16, 0, 0),
    end: new Date(2021, 8, 1, 17, 0, 0),
    title: "T shirt shredding design and cutting.",
    description:
      "Learn to turn any regular shirt into wearable art with just scissors.",
  },
  {
    start: new Date(2021, 8, 2, 6, 0, 0),
    end: new Date(2021, 8, 2, 7, 0, 0),
    title: "Bacon Bloody Sunrise",
    description:
      "Come drink bacon bloody marys and watch the sun rise from the summit of BRC.",
  },
  {
    start: new Date(2021, 8, 2, 21, 0, 0),
    end: new Date(2021, 8, 3, 0, 0, 0),
    title: "Funky Drunky Groove Party",
    description:
      "Shake-loose your caboose on a magical ride through the far reaches of the FUNKIVERSE as we explore a wild selection of unapologetically filthy jams.",
  },
  {
    start: new Date(2021, 8, 2, 14, 0, 0),
    end: new Date(2021, 8, 2, 15, 0, 0),
    title: "PolyJAMorous Music Theory Class",
    description:
      "Come join PolyJAMorous as we explore the world of music theory.  All levels welcome!",
  },
  {
    start: new Date(2021, 8, 4, 14, 0, 0),
    end: new Date(2021, 8, 4, 15, 0, 0),
    title: "PolyJAMorous Music Theory Class",
    description:
      "Come join PolyJAMorous as we explore the world of music theory.  All levels welcome!",
  },
  {
    start: new Date(2021, 7, 30, 18, 0, 0),
    end: new Date(2021, 7, 30, 19, 30, 0),
    title: "Sunset Jam Sesh",
    description:
      "Join us at sunset everyday for a jam session at the PolyJamerous tent. There will be musical instruments of all kinds!",
  },
  {
    start: new Date(2021, 7, 31, 18, 0, 0),
    end: new Date(2021, 7, 31, 19, 30, 0),
    title: "Sunset Jam Sesh",
    description:
      "Join us at sunset everyday for a jam session at the PolyJamerous tent. There will be musical instruments of all kinds!",
  },
  {
    start: new Date(2021, 8, 1, 18, 0, 0),
    end: new Date(2021, 8, 1, 19, 30, 0),
    title: "Sunset Jam Sesh",
    description:
      "Join us at sunset everyday for a jam session at the PolyJamerous tent. There will be musical instruments of all kinds!",
  },
  {
    start: new Date(2021, 8, 2, 18, 0, 0),
    end: new Date(2021, 8, 2, 19, 30, 0),
    title: "Sunset Jam Sesh",
    description:
      "Join us at sunset everyday for a jam session at the PolyJamerous tent. There will be musical instruments of all kinds!",
  },
  {
    start: new Date(2021, 8, 3, 18, 0, 0),
    end: new Date(2021, 8, 3, 19, 30, 0),
    title: "Sunset Jam Sesh",
    description:
      "Join us at sunset everyday for a jam session at the PolyJamerous tent. There will be musical instruments of all kinds!",
  },
  {
    start: new Date(2021, 8, 4, 18, 0, 0),
    end: new Date(2021, 8, 4, 19, 30, 0),
    title: "Sunset Jam Sesh",
    description:
      "Join us at sunset everyday for a jam session at the PolyJamerous tent. There will be musical instruments of all kinds!",
  },
  {
    start: new Date(2021, 7, 30, 12, 0, 0),
    end: new Date(2021, 7, 30, 16, 0, 0),
    title: "Black Rock City Summit Photo Opp",
    description:
      "Congratulations!  You made it all the way to the summit of Black Rock City at 3907 feet above sea level.  Mark this amazing accomplishment by stopping by and having your photo taken (or take your own) with the summit as your backdrop.  Photos taken by 3907 will be available after the burn on our facebook page.",
  },
  {
    start: new Date(2021, 7, 31, 12, 0, 0),
    end: new Date(2021, 7, 31, 16, 0, 0),
    title: "Black Rock City Summit Photo Opp",
    description:
      "Congratulations!  You made it all the way to the summit of Black Rock City at 3907 feet above sea level.  Mark this amazing accomplishment by stopping by and having your photo taken (or take your own) with the summit as your backdrop.  Photos taken by 3907 will be available after the burn on our facebook page.",
  },
  {
    start: new Date(2021, 8, 1, 12, 0, 0),
    end: new Date(2021, 8, 1, 16, 0, 0),
    title: "Black Rock City Summit Photo Opp",
    description:
      "Congratulations!  You made it all the way to the summit of Black Rock City at 3907 feet above sea level.  Mark this amazing accomplishment by stopping by and having your photo taken (or take your own) with the summit as your backdrop.  Photos taken by 3907 will be available after the burn on our facebook page.",
  },
  {
    start: new Date(2021, 8, 2, 12, 0, 0),
    end: new Date(2021, 8, 2, 16, 0, 0),
    title: "Black Rock City Summit Photo Opp",
    description:
      "Congratulations!  You made it all the way to the summit of Black Rock City at 3907 feet above sea level.  Mark this amazing accomplishment by stopping by and having your photo taken (or take your own) with the summit as your backdrop.  Photos taken by 3907 will be available after the burn on our facebook page.",
  },
  {
    start: new Date(2021, 8, 3, 12, 0, 0),
    end: new Date(2021, 8, 3, 16, 0, 0),
    title: "Black Rock City Summit Photo Opp",
    description:
      "Congratulations!  You made it all the way to the summit of Black Rock City at 3907 feet above sea level.  Mark this amazing accomplishment by stopping by and having your photo taken (or take your own) with the summit as your backdrop.  Photos taken by 3907 will be available after the burn on our facebook page.",
  },
  {
    start: new Date(2021, 8, 4, 12, 0, 0),
    end: new Date(2021, 8, 4, 16, 0, 0),
    title: "Black Rock City Summit Photo Opp",
    description:
      "Congratulations!  You made it all the way to the summit of Black Rock City at 3907 feet above sea level.  Mark this amazing accomplishment by stopping by and having your photo taken (or take your own) with the summit as your backdrop.  Photos taken by 3907 will be available after the burn on our facebook page.",
  },
  {
    start: new Date(2021, 8, 5, 12, 0, 0),
    end: new Date(2021, 8, 5, 16, 0, 0),
    title: "Black Rock City Summit Photo Opp",
    description:
      "Congratulations!  You made it all the way to the summit of Black Rock City at 3907 feet above sea level.  Mark this amazing accomplishment by stopping by and having your photo taken (or take your own) with the summit as your backdrop.  Photos taken by 3907 will be available after the burn on our facebook page.",
  },
  {
    start: new Date(2021, 8, 4, 21, 0, 0, 0),
    end: new Date(2021, 8, 4, 23, 30, 0, 0),
    title: "Man Burn",
    global: true,
  },
  {
    start: new Date(2021, 8, 5, 18, 0, 0, 0),
    end: new Date(2021, 8, 5, 21, 30, 0, 0),
    title: "Temple Burn",
    global: true,
  }
];
