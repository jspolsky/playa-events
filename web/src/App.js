import React, { useState } from "react";

import { Calendar, Views, momentLocalizer } from "react-big-calendar";
import withDragAndDrop from "react-big-calendar/lib/addons/dragAndDrop";
import "react-big-calendar/lib/addons/dragAndDrop/styles.scss";
import Alert from "@material-ui/lab/Alert";
import Snackbar from "@material-ui/core/Snackbar";

import { Header } from "./components/Header.js";

import moment from "moment";

import { burningManDates, yearDefault } from "./dateFunctions.js";
import { BurnWeek } from "./components/BurnWeek.js";
import { initialRawEvents } from "./sampleEvents";
import { EventDialog } from "./components/EventDialog";
import { CheckEvent } from "./components/CheckEvent.js";

export const sum = (a, b) => {
  return a + b;
};

// Playa Events!

const localizer = momentLocalizer(moment);
const DraggableCalendar = withDragAndDrop(Calendar);
const [firstDay] = burningManDates(yearDefault());

// At startup, assign each raw event an id that can be
// used to find it later.
//
let identity = 0;
initialRawEvents.forEach((e) => {
  e.id = identity++;
});

// Convert rawEvents to calEvents
//
// rawEvents format looks like this:
//
// {
//   start: { h: 20, m: 0 },
//   end: { h: 2, m: 0 },
//   days: [1,2,4],
//   title: "Party",
//   description:
//     "Future Turtles Party",
// }
//
// This function:
//
// - explodes multi-day events into several separate events (in the example above, there are three instances of the event)
// - convert the start and end times to actual Dates
//
export const CalEventsFromRawEvents = (rawEvents) => {
  let calEvents = [];

  rawEvents.forEach((e) => {
    e.eventError = CheckEvent(e);

    e.days.forEach((d) => {
      calEvents.push({
        start: moment(firstDay)
          .add(d, "days")
          .add(e.start.h, "hours")
          .add(e.start.m, "minutes")
          .toDate(),
        end: moment(firstDay)
          .add(d + (e.end.h < e.start.h ? 1 : 0), "days")
          .add(e.end.h, "hours")
          .add(e.end.m, "minutes")
          .toDate(),
        title: e.title,
        id: e.id, // multiple calendar events may point to same raw event
        global: e.global,
        eventError: e.eventError,
      });
    });
  });

  return calEvents;
};

function App() {
  const [rawEvents, setRawEvents] = useState(initialRawEvents);
  const [showPopup, setShowPopup] = useState(false);
  const [eventForPopup, setEventForPopup] = useState({});
  const [editing, setEditing] = useState(false);
  const [globalError, setGlobalError] = useState(false);
  const [dblClickTimer, setDblClickTimer] = useState(null);

  const newEventTemplate = {
    title: "New event",
    type: "othr",
    atCamp: true,
    locationType: "other",
    newEvent: true, // this makes it so that a "cancel" doesn't save
    eventError: false,
    description: "",
    url: "",
  };

  const newEventFromGrid = ({ start, end }) => {
    const newEvent = {
      ...newEventTemplate,
      start: { h: start.getHours(), m: start.getMinutes() },
      end: { h: end.getHours(), m: end.getMinutes() },
      days: [moment(start).diff(firstDay, "days")],
      id: identity++,
    };

    setRawEvents([...rawEvents, newEvent]);
    setEventForPopup(newEvent);
    setShowPopup(true);
    setEditing(true);
  };

  const handleNewEvent = () => {
    const newEvent = {
      ...newEventTemplate,
      start: { h: 12, m: 0 },
      end: { h: 14, m: 0 },
      days: [1],
      id: identity++,
    };

    setRawEvents([...rawEvents, newEvent]);
    setEventForPopup(newEvent);
    setShowPopup(true);
    setEditing(true);
  };

  const drillDown = (event) => {
    // single click - show the event in static mode
    // double click - show the event for editing

    setEventForPopup(rawEvents.find((e) => e.id === event.id));
    setEditing(false);

    if (dblClickTimer) {
      clearTimeout(dblClickTimer);
      setDblClickTimer(null);
      setEditing(!eventForPopup.global); // but you can never edit global events
    }

    setDblClickTimer(
      setTimeout(() => {
        setShowPopup(true);
        setDblClickTimer(null);
      }, 200)
    );
  };

  const closeDrillDown = () => {
    setShowPopup(false);
  };

  function isValidDate(d) {
    return d instanceof Date && !isNaN(d);
  }

  const moveEvent = ({ event, start, end, isAllDay }) => {
    if (!isValidDate(start) || !isValidDate(end)) {
      // the resizing code can corrupt the new start date when events cross midnight
      // https://github.com/jquense/react-big-calendar/issues/1598
      return;
    }

    if (event.global) {
      // are you trying to change the time of the man burn? good luck with that buddy
      return;
    }

    setRawEvents(
      rawEvents.map((e) => {
        if (e.id === event.id) {
          let newDays = e.days;

          // only single-day events can be moved date-wise
          // otherwise you gotta use the dialog
          if (newDays.length === 1) {
            newDays[0] = moment(start).diff(moment(firstDay), "days");
          }

          const result = {
            ...e,
            start: { h: start.getHours(), m: start.getMinutes() },
            end: { h: end.getHours(), m: end.getMinutes() },
            days: newDays,
          };

          setGlobalError(CheckEvent(result));

          return result;
        } else {
          return e;
        }
      })
    );
  };

  const saveEvent = (rawEvent) => {
    setGlobalError(CheckEvent(rawEvent));

    setRawEvents(
      rawEvents.map((e) => {
        if (e.id === rawEvent.id) {
          return rawEvent;
        } else {
          return e;
        }
      })
    );
  };

  const deleteEvent = (rawEventId) => {
    setRawEvents(rawEvents.filter((x) => x.id !== rawEventId));
    setGlobalError(false);
  };

  const formatEvent = (event, start, end, isSelected) => {
    let bgColor = "#00203F";
    if (event.global) {
      bgColor = "#E4DDCD";
    }
    if (event.eventError) {
      bgColor = "#DC4156";
    }

    return {
      style: {
        backgroundColor: bgColor,
        color: event.global ? "black" : "white",
        fontSize: "0.8rem",
        fontWeight: 500,
        opacity: 0.7,
      },
    };
  };

  return (
    <div className="App">
      <Snackbar open={globalError !== false}>
        <Alert severity="error" elevation={6} variant="filled">
          {globalError}
        </Alert>
      </Snackbar>
      <EventDialog
        show={showPopup}
        close={closeDrillDown}
        event={eventForPopup}
        saveEvent={saveEvent}
        deleteEvent={deleteEvent}
        editing={editing}
        setEditing={setEditing}
      />
      <DraggableCalendar
        localizer={localizer}
        defaultDate={firstDay}
        defaultView={Views.WEEK}
        events={CalEventsFromRawEvents(rawEvents)}
        style={{ height: "100vh" }}
        views={{ agenda: true, week: BurnWeek }}
        formats={{
          timeGutterFormat: (date) => moment(date).format("ha"),
          dayFormat: (date) => moment(date).format("ddd, MMM D"),
          eventTimeRangeFormat: ({ start, end }) =>
            moment(start).format("h:mma"),
          eventTimeRangeStartFormat: ({ start, end }) =>
            moment(start).format("h:mma"),
          eventTimeRangeEndFormat: ({ start, end }) =>
            moment(end).format("—h:mma"),
          selectRangeFormat: ({ start, end }) =>
            moment(start).format("h:mma—") + moment(end).format("h:mma"),
        }}
        selectable
        onSelectEvent={drillDown}
        onSelectSlot={newEventFromGrid}
        showMultiDayTimes
        onEventDrop={moveEvent}
        onEventResize={moveEvent}
        eventPropGetter={formatEvent}
        components={{
          event: ({ event }) => {
            return (
              <span>
                {!!event.eventError ? "⚠️ " : ""}
                {event.title}
              </span>
            );
          },
          toolbar: (props) => {
            return (
              <Header title={props.label} handleNewEvent={handleNewEvent} />
            );
          },
        }}
      />
    </div>
  );
}

export default App;
