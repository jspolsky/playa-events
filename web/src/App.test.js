import { render, screen } from "@testing-library/react";
import App, { sum, CalEventsFromRawEvents } from "./App";
import { initialRawEvents } from "./sampleEvents";

test("1 + 1 = 2", () => {
  expect(sum(1, 1)).toBe(2);
});

test("CalEventsFromRawEvents", () => {
  expect(CalEventsFromRawEvents(initialRawEvents)).toMatchSnapshot();
});
