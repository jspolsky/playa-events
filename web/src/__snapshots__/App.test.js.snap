// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalEventsFromRawEvents 1`] = `
Array [
  Object {
    "end": 2021-08-31T06:00:00.000Z,
    "global": undefined,
    "id": 0,
    "start": 2021-08-31T00:00:00.000Z,
    "title": "East Village Shirtless Dance Party",
  },
  Object {
    "end": 2021-09-05T03:30:00.000Z,
    "global": true,
    "id": 1,
    "start": 2021-09-05T01:00:00.000Z,
    "title": "Man Burn",
  },
  Object {
    "end": 2021-09-06T01:00:00.000Z,
    "global": true,
    "id": 2,
    "start": 2021-09-05T22:00:00.000Z,
    "title": "Temple Burn",
  },
  Object {
    "end": 2021-08-31T16:00:00.000Z,
    "global": undefined,
    "id": 3,
    "start": 2021-08-31T15:00:00.000Z,
    "title": "Lazy workout",
  },
  Object {
    "end": 2021-09-01T16:00:00.000Z,
    "global": undefined,
    "id": 3,
    "start": 2021-09-01T15:00:00.000Z,
    "title": "Lazy workout",
  },
  Object {
    "end": 2021-09-02T16:00:00.000Z,
    "global": undefined,
    "id": 3,
    "start": 2021-09-02T15:00:00.000Z,
    "title": "Lazy workout",
  },
  Object {
    "end": 2021-09-03T16:00:00.000Z,
    "global": undefined,
    "id": 3,
    "start": 2021-09-03T15:00:00.000Z,
    "title": "Lazy workout",
  },
  Object {
    "end": 2021-09-04T16:00:00.000Z,
    "global": undefined,
    "id": 3,
    "start": 2021-09-04T15:00:00.000Z,
    "title": "Lazy workout",
  },
  Object {
    "end": 2021-09-01T03:59:00.000Z,
    "global": undefined,
    "id": 4,
    "start": 2021-08-31T22:00:00.000Z,
    "title": "Your Dreams Interpreted",
  },
  Object {
    "end": 2021-09-03T03:59:00.000Z,
    "global": undefined,
    "id": 4,
    "start": 2021-09-02T22:00:00.000Z,
    "title": "Your Dreams Interpreted",
  },
  Object {
    "end": 2021-09-02T03:59:00.000Z,
    "global": undefined,
    "id": 5,
    "start": 2021-09-01T22:00:00.000Z,
    "title": "Cook yer sausage",
  },
  Object {
    "end": 2021-09-02T22:00:00.000Z,
    "global": undefined,
    "id": 6,
    "start": 2021-09-02T20:00:00.000Z,
    "title": "Billion Bunny March",
  },
  Object {
    "end": 2021-09-04T03:59:00.000Z,
    "global": undefined,
    "id": 7,
    "start": 2021-09-03T22:00:00.000Z,
    "title": "AP Calculus Study Group",
  },
  Object {
    "end": 2021-09-04T06:00:00.000Z,
    "global": undefined,
    "id": 8,
    "start": 2021-09-04T04:45:00.000Z,
    "title": "Impossibility space",
  },
  Object {
    "end": 2021-08-31T04:45:00.000Z,
    "global": undefined,
    "id": 9,
    "start": 2021-08-31T04:00:00.000Z,
    "title": "Midnight coffee",
  },
  Object {
    "end": 2021-09-01T04:45:00.000Z,
    "global": undefined,
    "id": 9,
    "start": 2021-09-01T04:00:00.000Z,
    "title": "Midnight coffee",
  },
  Object {
    "end": 2021-09-02T04:45:00.000Z,
    "global": undefined,
    "id": 9,
    "start": 2021-09-02T04:00:00.000Z,
    "title": "Midnight coffee",
  },
  Object {
    "end": 2021-09-03T04:45:00.000Z,
    "global": undefined,
    "id": 9,
    "start": 2021-09-03T04:00:00.000Z,
    "title": "Midnight coffee",
  },
  Object {
    "end": 2021-09-04T04:45:00.000Z,
    "global": undefined,
    "id": 9,
    "start": 2021-09-04T04:00:00.000Z,
    "title": "Midnight coffee",
  },
]
`;
